src/ui/QuerySuggestPreview/QuerySuggestPreview.ts

```diff
@@ -244,6 +258,21 @@ export class QuerySuggestPreview extends Component implements IComponentBindings
     });
   }
 
+  private addOnClickListener(results: HTMLElement[]) {
+    results.forEach(result => {
+      const rank = results.indexOf(result).toString();
+      result.setAttribute('coveo-preview-rank', rank);
```

Comment:
I think you could do without this attribute. Just pass in the rank from the closure.

```
const rank = results.indexOf(result).toString();
this.bind.on(result, 'click' ()=>{
   this.handleOnClick(rank, element)
})
```